### DISCUSSION TITLE
Blacklist Line Limit Workaround not working

### DISCUSSION THREAD

**Discussion by abaler** [jertel/elastalert2#1038] [Q&A] [answered]
Hello!
I saw in [blacklist line limit #3015](https://github.com/Yelp/elastalert/issues/3015) that separating the blacklist into separate files that are shorter than 1024 supposedly works. 
Unfortunately for me it did not work, currently i have two files each 300 lines, but if i add a 3rd one of the same length it throws an error:

`ERROR:elastalert:Error running query: ['Failed to parse query [saving_grace:"microsoft:applicationinspector" OR saving_grace:"microsoft:ms-dos" OR saving_grace:"microsoft:outlook_2013_rt" OR saving_grace:"gemalto:safenet_authentication_service_end_user_software_tools_for_windows" OR saving_grace:"microsoft:onenote_for_mac" OR saving_grace:"microsoft:microsoft_data_access_components" OR saving_grace:"gnu:gettext" OR saving_grace:"hp:software_perl" OR saving_grace:"bashgaming:bingo_bash_free_bingo_casino" OR saving_grace:"altiris:patch_management_core_solution_for_linux" OR saving_grace:"microsoft:powershell_extension" OR saving_grace:"microsoft:365_apps" OR saving_grace:"microsoft:debug_interface_access_software_development_kit" OR saving_grace:"biscom:faxcom_suite_for_windows_client" OR saving_grace:"microsoft:windows-9x" OR saving_grace:"microsoft:netmeeting" OR saving_grace:"openssl_project:openssl" OR saving_grace:"vanilla_perl_project:strawberry_perl" OR saving_grace:"adobe:flash_player_installer" OR saving_grace:"pgp:desktop_for_windows" OR ... (44592 characters removed)`

What am i doing wrong? Or is there another way to bypass the line limit?

**Comment by jertel** [comment, marked as answer]
Splitting the large file into smaller files but then referencing all of those smaller files in a single rule is functionally no different than keeping them in the same large file. 

I believe the users in the referenced discussion are splitting the files into 1024 lines and then creating one rule per file to get around the Elastic query string term limit.

However, since you only have 3 files, totaling just 900, that falls under the 1024 limit. So I suspect that you are actually running into a different problem, such as one of the files containing slash / or double quote " characters in those lines, or there is some other content/length of the line causing an issue. If a line starts with / and also ends with / then it must be a valid regular expression (regex). If a line contains a double quote " character then it may need to be escaped.

**Comment by abaler** [comment]
Unfortunately, I have more than 3 files, more like 23, each with 300 lines, but it failed at the third one. 
And i did just check, there are certain lines that have this issue of containing a slash or a double quote. Thank you for the tip.
I guess I will go with the multiple rules to bypass the issue.
Thank you for the help!
